### PR TITLE
Update tests to match SHA1 cutoff date for key signatures.

### LINKED ISSUE #1961: Tests fail after 2024-01-19
## Description

When building the `rnp-0.16.2` openSUSE package after 2024-01-19, three tests fail

## Steps to Reproduce

1. on Debian or openSUSE do
2. `osc co openSUSE:Factory/rnp && cd $_`
3. `osc build --vm-type=kvm --noservice --clean --build-opt=--vm-custom-opt="-rtc base=2024-01-19T12:00:00" standard`

## Expected Behavior

Tests should keep passing in future

## Actual Behavior

```
 The following tests FAILED: 
         75 - rnp_tests.test_key_add_userid (Failed)
        207 - rnp_tests.test_ffi_security_profile (Failed)
        252 - cli_tests-EncryptElgamal (Failed)
 Errors while running CTest
```

The failure is probably from `src/lib/sec_profile.cpp` SHA-1 depecation

Background:
As part of my work on reproducible builds for openSUSE, I check that software still gives identical build results in the future.
The usual offset is +16 years, because that is how long I expect some software will be used in some places.
This showed up failing tests in our package build.
See https://reproducible-builds.org/ for why this matters.

### LINKED ISSUE #2182: test_key_add_userid and test_ffi_security_profile failure
`test_key_add_userid` and test_ffi_security_profile fail in similar ways 0.17.0.

This is probably related to an update of a dependency in the distribution, we need to yet determine which one.

current log: https://build.opensuse.org/package/live_build_log/security:privacy/rnp/openSUSE_Tumbleweed/x86_64
current sources https://build.opensuse.org/package/show/openSUSE:Factory/rnp

````
[  127s]         Start  76: rnp_tests.test_key_add_userid
[  127s]  74/260 Test  #76: rnp_tests.test_key_add_userid .................................................***Failed    0.11 sec
[  127s] Note: Google Test filter = rnp_tests.test_key_add_userid
[  127s] [==========] Running 1 test from 1 test suite.
[  127s] [----------] Global test environment set-up.
[  127s] [----------] 1 test from rnp_tests
[  127s] [ RUN      ] rnp_tests.test_key_add_userid
[  127s] /home/abuild/rpmbuild/BUILD/rnp-v0.17.0/src/tests/key-add-userid.cpp:75: Failure
[  127s] Expected equality of these values:
[  127s]   (base_expiry)
[  127s]     Which is: 1234567890
[  127s]   (key->expiration())
[  127s]     Which is: 0
[  127s] [  FAILED  ] rnp_tests.test_key_add_userid (94 ms)
[  127s] [----------] 1 test from rnp_tests (94 ms total)
[  127s] 
[  127s] [----------] Global test environment tear-down
[  127s] [==========] 1 test from 1 test suite ran. (94 ms total)
[  127s] [  PASSED  ] 0 tests.
[  127s] [  FAILED  ] 1 test, listed below:
[  127s] [  FAILED  ] rnp_tests.test_key_add_userid
[  127s] 
[  127s]  1 FAILED TEST
[  127s] 
````

````
[  150s] 212/260 Test #212: rnp_tests.test_ffi_security_profile ...........................................***Failed    0.52 sec
[  150s] Note: Google Test filter = rnp_tests.test_ffi_security_profile
[  150s] [==========] Running 1 test from 1 test suite.
[  150s] [----------] Global test environment set-up.
[  150s] [----------] 1 test from rnp_tests
[  150s] [ RUN      ] rnp_tests.test_ffi_security_profile
[  150s] /home/abuild/rpmbuild/BUILD/rnp-v0.17.0/src/tests/ffi.cpp:5956: Failure
[  150s] Expected equality of these values:
[  150s]   (from)
[  150s]     Which is: 1705629600
[  150s]   (1547856000)
[  150s]     Which is: 1547856000
[  150s] [  FAILED  ] rnp_tests.test_ffi_security_profile (507 ms)
[  150s] [----------] 1 test from rnp_tests (507 ms total)
[  150s] 
[  150s] [----------] Global test environment tear-down
[  150s] [==========] 1 test from 1 test suite ran. (507 ms total)
[  150s] [  PASSED  ] 0 tests.
[  150s] [  FAILED  ] 1 test, listed below:
[  150s] [  FAILED  ] rnp_tests.test_ffi_security_profile
[  150s] 
[  150s]  1 FAILED TEST
[  150s] 
````

### PR DESCRIPTION
**Pull request by ni4**
Fixes #1961 
Fixes #2182

### CODE REVIEW

**Comment by ni4** [comment]
Not sure why codecov is not happy here.

**Review by ronaldtse — APPROVED** [review]

File: src/tests/cli_tests.py [hunk 1]
```diff
@@ -4882,7 +4886,7 @@ def test_generate_elgamal_key1024_in_gpg_and_encrypt(self):
         cmd = EncryptElgamal.GPG_GENERATE_DSA_ELGAMAL_PATTERN.format(1024, 1024, self.gpg.userid)
         self.operation_key_gencmd = cmd
         # Will not fail till 2024 since 1024-bit DSA key uses SHA-1 as hash.
-        self._encrypt_decrypt(self.gpg, self.rnp)
```
Comment (maxirmx) [inline src/tests/cli_tests.py:4885]: It looks like this test is doing nothing now. Am I wrong ?

**Review by maxirmx — CHANGES_REQUESTED** [review]
With a call to ```encrypt_decrypt``` commented ```test_generate_elgamal_key1024_in_gpg_and_encrypt(self)```  can be removed.

I did not try to understabd the logic, just a formal note.

**Review by antonsviridenko — APPROVED** [review]

**Review by maxirmx — APPROVED** [review]

**Comment by ni4** [comment]
Let's merge it as codecov diff report somehow glitched.

**Pull request merged**